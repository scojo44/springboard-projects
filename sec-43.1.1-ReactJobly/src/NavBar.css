.NavBar {
  display: flex;
  align-items: center;
  color: white;
  background-color: indigo;
  padding: .25rem;
}
.NavBar a:first-child { /* Logo link */
  flex: 1;
  font-style: italic;
  font-size: larger;
}
.NavBar a {
  font-weight: bold;
  padding: 0 1rem;
}
.NavBar a:link,
.NavBar a:visited {
  color: white;
}