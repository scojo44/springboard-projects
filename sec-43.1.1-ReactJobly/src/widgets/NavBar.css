.NavBar {
  display: flex;
  align-items: center;
  color: white;
  background-color: cornflowerblue;
  padding: .5rem;
}
.NavBar a.active {
  font-size: 1.25rem;
}
.NavBar a {
  font-weight: bold;
  padding: 0 1rem;
  text-decoration: none;
}
.NavBar a:link,
.NavBar a:visited {
  color: white;
}
.NavBar a:first-child { /* Logo link */
  flex: 1;
  font-style: italic;
  font-size: 1.25rem;
}
