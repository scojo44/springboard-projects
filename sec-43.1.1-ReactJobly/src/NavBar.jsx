import React, { useState } from 'react'
import {Link, NavLink} from 'react-router-dom'
import './NavBar.css'
import Alert from './Alert';

export default function NavBar() {
  const [loggedOut, setLoggedOut] = useState(false);
  return (
    <>
    <nav className="NavBar">
      <Link to="/">Jobly</Link>
      <NavLink to="/companies">Companies</NavLink>
      <NavLink to="/jobs">Jobs</NavLink>
      <NavLink to="/login">Log In</NavLink>
      <NavLink to="/signup">Sign Up</NavLink>
      <NavLink to="/profile">Profile</NavLink>
      <Link onClick={() => setLoggedOut(true)}>Log Out</Link>
    </nav>
    {loggedOut && <Alert type="success" messages={['Logout Clicked!']} />}
    </>
  )
}
