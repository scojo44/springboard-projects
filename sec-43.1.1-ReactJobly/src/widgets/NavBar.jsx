import React, {useContext} from 'react'
import {Link, NavLink} from 'react-router-dom'
import CurrentUserContext from '../CurrentUserContext';
import './NavBar.css'

export default function NavBar({logout}) {
  const user = useContext(CurrentUserContext);

  return (
    <>
    <nav className="NavBar">
      <Link to="/">Jobly</Link>
      <NavLink to="/companies">Companies</NavLink>
      <NavLink to="/jobs">Jobs</NavLink>
      {user
        ? <>
            <NavLink to="/profile">Profile</NavLink>
            <Link onClick={logout}>Log Out</Link>
          </>
        : <>
            <NavLink to="/signup">Sign Up</NavLink>
            <NavLink to="/login">Log In</NavLink>
          </>
      }
    </nav>
    </>
  );
}
