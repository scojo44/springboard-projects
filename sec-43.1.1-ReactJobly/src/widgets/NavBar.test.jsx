import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import NavBar from './NavBar';

describe('NavBar Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
