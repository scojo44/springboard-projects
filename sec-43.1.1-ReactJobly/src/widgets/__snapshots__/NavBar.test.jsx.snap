// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBar"
  >
    <a
      href="/"
    >
      Jobly
    </a>
    <a
      class=""
      href="/signup"
    >
      Sign Up
    </a>
    <a
      class=""
      href="/login"
    >
      Log In
    </a>
  </nav>
</DocumentFragment>
`;
